src/main/java/core/basesyntax/dao/impl/FileWriterImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.FruitFileWriter;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class FileWriterImpl implements FruitFileWriter {
+    @Override
+    public void write(String report, String filename) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("An error occurred while writing to a file: " + filename, e);
```

Comment:
Consider providing more specific error messages or handling different types of IOExceptions separately to give more context about what went wrong (e.g., file not found, access denied).